src/services/analytics/PostHogClient.ts

```diff
@@ -0,0 +1,42 @@
+import { PostHog } from "posthog-node"
+import * as vscode from "vscode"
+
+class PostHogClient {
+	private static instance: PostHogClient
+	private client: PostHog
+	private distinctId: string = vscode.env.machineId
+
+	private constructor() {
+		this.client = new PostHog("phc_5WnLHpYyC30Bsb7VSJ6DzcPXZ34JSF08DJLyM7svZ15", {
+			host: "https://us.i.posthog.com",
```

Comment:
nit: Consider placing the URL and key into a `const`, as it seems that is also being repeated in the web-view comp.